### DISCUSSION TITLE
Can not get ImageInfo in Ubuntu 20.04.1 LTS

### DISCUSSION THREAD

**Discussion by drayvs** [ImageMagick/ImageMagick#3036] [Help]
Here is my situation, I have 2 files (for example):
- one is 1168x1110, with Bit depth of 4, size 97.8KB, 006.png
- another is 1080x21095, with Bit depth of 8, 3.29MB, 001.png

I use this script - imageinfo to get information about image, it works on ImageMagick library, you can download it here -https://file.io/paKWyhYqkLrM
It was written by  Brendt Wohlberg, back in 1999.

With this command "imageinfo --format --size --width --height --md5hash 001.png"

I get 0 information, nothing about 001.png, and it work just fine with 006.png, I get info like so
'006.png\tPNG   100259   1168   1110   b0a64599c53bb4bee64b5789fab53eae'

If use file 001.png it works just fine:
001.png: PNG image data, 1080 x 21095, 8-bit colormap, non-interlaced

So issue is obviously is with imagemagick library, can someone help me with this?

**Comment by snibgo** [comment]
What is the "imageinfo" command? That is not part of ImageMagick, as far as I know.

**Reply by drayvs** [reply]
> What is the "imageinfo" command? That is not part of ImageMagick, as far as I know.

It uses Magick's library, which isn't working at the moment. This part in particular:
GetExceptionInfo(&exception);
  image_info=CloneImageInfo((ImageInfo *) NULL);
  strcpy(image_info->filename, filename);
  if (readimage)
    image = ReadImage(image_info, &exception);
  else
    image = PingImage(image_info, &exception);
  if (!image) {
    /* Use MagickWarning rather than MagickError beacuse the latter
       behaves incorrectly when an empty file is encountered */
    MagickWarning(exception.severity,exception.reason,exception.description);
    DestroyImageInfo(image_info);
    DestroyExceptionInfo(&exception);
    MagickCoreTerminus();
    exit(5);
  }

**Comment by snibgo** [comment]
Okay, so "imageinfo" is a program written by you or someone else.

Debugging a program without seeing the source code is difficult. Does your "if (!image) {...}" code get executed?

Does it write anything to stderr?

Does "magick 001.png -verbose info:" work?

Your use of "exception" looks wrong. Does your program build without warnings?

**Reply by drayvs** [reply]
I put link to this script (source is also there).
I have nothing in STDERR, script just exits with code 5 (exit(5)), there isn't any errors or warnings either.

As for your command: "magick 001.png -verbose info"
It doesn't work, this is what I get:
magick: no decode delegate for this image format `PNG' @ error/constitute.c/ReadImage/572.

**Reply by fmw42** [reply]
What do you get from 

`magick -version`

**Reply by drayvs** [reply]
> What do you get from
> 
> `magick -version`

Version: ImageMagick 7.0.10-54 Q16 x86_64 2020-12-26 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in):

**Reply by fmw42** [reply]
You have no delegates such as for PNG, JPG, TIFF, etc.   How did you install Imagemagick? You should install from a binary not source. With a binary you get most if not all the delegates you need. With source you have to install each delegate before installing ImageMagick. 

I suspect something was wrong with your install or you just installed the source code. Perhaps see https://askubuntu.com/questions/1042436/how-to-install-delegate-libraries-for-image-magick-7-0-7

**Reply by drayvs** [reply]
Is there a way to delete it all and correctly install it? Or in any way I will have to install those dev libraries?

**Reply by fmw42** [reply]
`make uninstall`
"Remove all files from the system which are (or would be) installed by sudo make install using the current configuration. Note that this target is imperfect for PerlMagick since Perl no longer supports an uninstall target."

Sorry, I do not use Linux and know nothing about installation on Ubuntu.

See

https://imagemagick.org/script/download.php

and the reference I sent before.

**Reply by drayvs** [reply]
Phew, this kinda helped, i got rid of imageinfo, and got almost all information about image directly via magick's identify, but encountered another problem, as far I understand magick doesn't know how to get md5hash of the image thus I tried using signature, but it has its problems - .gif images have insane signature which breaks my script.

Can I somehow get md5hash of images with magick?
